### ISSUE TITLE
Some BC10 and NB02 translated stands have only one layer = 2 row (no row with layer = 1)

### ISSUE THREAD

**Issue by pedrogit** [edwardsmarc/CASFRI#330]
To find all the BC10 ones:

```
WITH cas_ids AS (
  SELECT cas_id, string_agg(layer::text, ', ' ORDER BY layer) layer
  FROM casfri50.lyr_all
  WHERE left(cas_id, 4) = 'BC10'
  GROUP BY cas_id
  HAVING string_agg(layer::text, ', ' ORDER BY layer) = '2'
  LIMIT 10
)
SELECT cas_id, l1_species_cd_1, l1_species_cd_2, l1_species_cd_3, l2_species_cd_1, l2_species_cd_2, l2_species_cd_3
FROM cas_ids, rawfri.bc10 rf
WHERE right(cas_ids.cas_id, 7)::int = rf.ogc_fid;
```

Here is a specific one:

```
SELECT * FROM casfri50.lyr_all
WHERE cas_id = 'BC10-VEG_COMP_LYR_L2-xxx082C096-0008298363-0893303';
```
Just replace BC10 with NB02 to find the ones in NB02. There seems to be only one in NB02: "NB02-GEONB_FOREST_FO-xxxxxxxxxx-0000350728-0603640"

**Comment by pedrogit** [comment]
I found the explanation for NB02. It maight be the same explanation for BC10.

The NB02 row has info only for species _4. When computing the layer number, this attribute is considered so that it get layer = 1. However TT_CreateMappingView() only consider the three first species attribute to decide if a row is translated or not and in this case it's not.

In brief a layer = 1 row should be produced but it's not.

**Comment by pedrogit** [comment]
I think we need to improve TT_CreateMappingView() so it takes more attribute into consideration when filtering out and so it's able to handle more specific filters for specific onventories (to also fix #329 and #325).

**Comment by edwardsmarc** [comment]
BC issue could be related to #331. I'll fix the translation and retest for this issue.

**Comment by pedrogit** [comment]
The explanation for BC10 is that the two rows exists but they are assigned a different cas_id. One is "BC10-VEG_COMP_LYR_L2-xxx082C096-0008298363-0893303" and the other is "BC10-VEG_COMP_LYR_L1-xxx082C096-0008298363-0893303".

**Comment by edwardsmarc** [comment]
Loaded table needs to have only one source_filename attribute. Not one per layer. Edit conversion script to fix this and change the src_filename mapping in attribute_dependencies.

**Comment by pedrogit** [comment]
I think only fixing the attribute_dependencies would be sufficient.

**Comment by edwardsmarc** [comment]
Yes we could just fix the attribute dependencies table, however that will lead to cas_ids that all point to the BC10-VEG_COMP_LYR_L1 as being the source table which is not true.

Fixing the loading script would mean the cas_id for every row would be something like BC10-VEG_COMP. This would link the cas_id to the original polygon table from the geodatabase which I think is what we want. This would break the tables though until the BC data was re-loaded, and it would require a new release.

**Comment by pedrogit** [comment]
BC was fixed by a fix for #344
https://github.com/edwardsmarc/CASFRI/commit/e000c2ccec554899d95a8e24387f0e5b59dc6f2f

**Comment by pedrogit** [comment]
Apparently the problem is completely fixed!
